internal/commands/scan.go

```diff
@@ -687,13 +699,31 @@ func updateProject(
 		logger.PrintIfVerbose("Updating project tags")
 		projModel.Tags = createTagMap(projectTags)
 	}
+	if applicationId != "" {
+		logger.PrintIfVerbose("Updating project applicationIds")
+		projModel.ApplicationIds = createApplicationIds(applicationId, projModelResp.ApplicationIds)
+	}
 	err = projectsWrapper.Update(projectID, &projModel)
 	if err != nil {
 		return "", errors.Errorf("%s: %v", failedUpdatingProj, err)
 	}
 	return projectID, nil
 }
 
+func createApplicationIds(applicationId string, existingApplicationIds []string) []string {
+	if hasElements(existingApplicationIds) {
+		if !contains(existingApplicationIds, applicationId) {
+			existingApplicationIds = append(existingApplicationIds, applicationId)
+		}
+		return existingApplicationIds
+	}
+	return []string{applicationId}
+}
+
+func hasElements(items []string) bool {
+	return items != nil && len(items) > 0
```

Comment:
Is len() throw error if nil? I think that in Go it'll return 0. (verify please)
If so, we don't need this method and it's better to have in the code if len(item) > 0.
if we do want this method, let's have an extension class for it

Reply:
You were right :-)